tests/test_controller.py

```diff
@@ -80,6 +85,39 @@
 from problem_details import *
 from viaf import MockVIAFClient
 
+def unauthenticated_request_context(f):
+    """A decorator for a test that runs in a simple request context
+    with no authentication.
+
+    This is useful if a test runs in one unauthenticated request that
+    does nothing special.
+    """
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        from app import app
+        with app.test_request_context('/'):
+            return f(*args, **kwargs)
+    return decorated
+
+def authenticated_request_context(f):
+    """A decorator for a test that runs in a simple request context
+    with valid authentication.
+
+    This is useful if a test runs in one authenticated request that
+    does nothing special.
+    """
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        from app import app
+
+        secret = args[0].client.shared_secret.encode('utf8')
+        valid_auth = 'Bearer '+ base64.urlsafe_b64encode(secret)
+        headers = { 'Authorization' : valid_auth }
+        with app.test_request_context('/', headers=headers):
+            return f(*args, **kwargs)
+    return decorated
+
+
```

Comment:
I moved these annotators out of TestIntegrationClientController so they could be used in other tests. They can't completely replace the need for the `authenticated_request` context manager, but when it is possible to use them, the test code is simpler.

Reply:
So both of these decorators aren't currently being used yet?

Reply:
They are used, but only on tests where the entire test runs in a single request where the only interesting thing about the request is "it's authenticated".